pallets/manta-sbt/src/lib.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright 2020-2023 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # MantaSBT Module
+//!
+//! MantaSBT creates non-transferable nfts as unspendable UTXOs
+//!
+//! ## Overview
+//!
+//! Uses `pallet-asset-manager` to store SBT metadata. Ownership is recorded as a corresponding UTXO.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(doc_cfg, feature(doc_cfg))]
+#![forbid(rustdoc::broken_intra_doc_links)]
+
+extern crate alloc;
+
+use alloc::{boxed::Box, vec, vec::Vec};
+use frame_support::{
+    pallet_prelude::*,
+    traits::{Currency, ExistenceRequirement, ReservableCurrency, StorageVersion},
+    transactional, PalletId,
+};
+use frame_system::pallet_prelude::*;
+use manta_primitives::assets::{AssetMetadata, SbtBound, UpdateMetadata};
+use manta_support::manta_pay::{
+    id_from_field, AssetType, PostToLedger, StandardAssetId, TransferPost,
+};
+use sp_runtime::{
+    traits::{AccountIdConversion, One},
+    ArithmeticError,
+};
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmark;
+
+// One in encoded form, used to check that value input in `ToPrivate` post is one
+const ENCODED_ONE: [u8; 16] = 1u128.to_le_bytes();
+
+/// Type alias for currency balance.
+pub type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// MantaSBT Pallet
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+    /// Pallet
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    /// The module configuration trait.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+
+        /// Balance Type
+        type Balance: Default + Member + Parameter + TypeInfo;
```

Comment:
Not sure why add new `Balance` type here.

Reply:
Yeah idk, must  of been leftover.

You are correct is unneeded